### PR TITLE
Boost cleanup from build files and a few other c++11 replacements for old usages of boost 

### PR DESCRIPTION
**Pull request by davidlange6**
#### PR description:

Boost cleanup from build files and a few other c++11 replacements for old usages of boost 

#### PR validation:

compiles and workflows run

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

File: DataFormats/Common/interface/Ref.h [hunk 1]
```diff
@@ -117,7 +117,6 @@ Ref: A template for a interproduct reference to a member of a product_.
 #include "boost/type_traits.hpp"
 #include "boost/mpl/has_xxx.hpp"
 #include "boost/utility/enable_if.hpp"
-
```
Comment (Dr15Jones) [inline DataFormats/Common/interface/Ref.h]: Why no boost change here?

Comment (davidlange6) [reply DataFormats/Common/interface/Ref.h]: sorry - I should not have left this space change in my PR - I am trying to change enable_if but I didn't manage to succeed yet.

Comment (Dr15Jones) [reply DataFormats/Common/interface/Ref.h]: Do you plan on reverting this or leave it as is?

**Comment by perrotta** [comment]
> Comparison Summary:
> 
> * No significant changes to the logs found
> * Reco comparison results: 10 differences found in the comparisons

After some while, the reproducibility issue which makes fluctuate the number of converted photons, apparently only in wf 11634 is reappearing, see also #28072 (and several other examples from PRs tested on Sep 21 and 22: it seemed to have disappeared after that, but...)

**Comment by mommsen** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/28089	524d084 
- technical
- jenkins tests pass; it looks like a significant code base recompilation still leads to occasional unrelated differences in wf 11634 as noticed in https://github.com/cms-sw/cmssw/pull/28089#issuecomment-536686790

**Comment by Martin-Grunewald** [comment]
+1

**Comment by Dr15Jones** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by pohsun** [comment]
+1

**Comment by santocch** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

@rekovic please have a look

**Pull request merged**